tests/test_setup.py

```diff
@@ -175,6 +183,99 @@ def test_validate_platform_config(self):
             assert 'platform_conf' in self.hass.config.components
             assert not config['platform_conf']  # empty
 
+    def test_validate_platform_config_2(self):
+        """Test component PLATFORM_SCHEMA_BASE prio over PLATFORM_SCHEMA."""
+        platform_schema = PLATFORM_SCHEMA.extend({
+            'cheers': str,
+            'hello': str,
+        })
+        platform_schema_base = PLATFORM_SCHEMA.extend({
+            'hello': str,
+        })
+        loader.set_component(
+            self.hass,
+            'platform_conf',
+            MockModule('platform_conf',
+                       platform_schema=platform_schema,
+                       platform_schema_base=platform_schema_base))
+
+        loader.set_component(
+            self.hass,
+            'platform_conf.whatever',
+            MockPlatform('whatever',
+                         platform_schema=platform_schema))
+
+        with assert_setup_component(0):
+            assert setup.setup_component(self.hass, 'platform_conf', {
+                'platform_conf': {
+                    'hello': 'world',
+                    'invalid': 'extra',
+                }
+            })
+
+        self.hass.data.pop(setup.DATA_SETUP)
+        self.hass.config.components.remove('platform_conf')
+
+        with assert_setup_component(1):
+            assert setup.setup_component(self.hass, 'platform_conf', {
+                'platform_conf': {
+                    'platform': 'whatever',
+                    'hello': 'world',
+                },
+                'platform_conf 2': {
+                    'platform': 'whatever',
+                    'hello': 'world',
+                    'cheers': 'mate'
+                }
+            })
+
+        self.hass.data.pop(setup.DATA_SETUP)
+        self.hass.config.components.remove('platform_conf')
+
+    def test_validate_platform_config_3(self):
+        """Test fallback to component PLATFORM_SCHEMA."""
```

Comment:
When are we falling back to component platform schema? It looks like the platform has a platform schema.